app/controllers/users_controller.rb

```diff
@@ -16,7 +14,19 @@ def create
 
   private
 
+  def user_from_params
+    email = user_params.delete(:email)
+    username = user_params.delete(:username)
+    password = user_params.delete(:password)
+
+    Clearance.configuration.user_model.new(user_params).tap do |user|
+      user.email = email
+      user.username = username
+      user.password = password
+    end
+  end
+
   def user_params
-    params.require(:user).permit(:username, :email, :password, :password_confirmation)
+    params[Clearance.configuration.user_parameter] || Hash.new
```

Comment:
You could use `params.fetch(Clearance.configuration.user_parameter, Hash.new)` here.

If this was a normal `Hash` there wouldn't be much of a difference, but in this case `params` is an instance of [`ActionController::Parameters`](http://api.rubyonrails.org/classes/ActionController/Parameters.html). `ActionController::Parameters#fetch` will make sure any Hash you get back from `#fetch` is converted to an instance of `ActionController::Parameters`, which means you can call methods like `#require` and `#permit` on it.